lib/kafka/protocol/message.rb

```diff
@@ -89,6 +89,10 @@ def self.decode(decoder)
           timestamp = nil
         when 1
           timestamp = message_decoder.int64
+
+          # If the timestamp is set to zero (start of UNIX epoch) there's clearly
```

Comment:
@dasch FYI that this can happen if you do an in place upgrade from 0.9.X to 0.10.X or onwards - old messages get zerod out (iirc)

Reply:
Ah! I'll fix the comment.